src/main/java/refinedstorage/tile/TileDestructor.java

```diff
@@ -125,23 +125,36 @@ public void updateNode() {
 
                 if (frontStack != null) {
                     if (IFilterable.canTake(itemFilters, mode, compare, frontStack)) {
-                        List<ItemStack> drops = frontBlockState.getBlock().getDrops(worldObj, front, frontBlockState, 0);
+                        if(!upgrades.hasUpgrade(ItemUpgrade.TYPE_SILK_TOUCH)) {
```

Comment:
is that different if branch really needed? couldn't you just insert have different contents in `drops` depending on the mode?